### PR TITLE
Fixes merch form for bands and guests

### PR DESCRIPTION
**Pull request by RobRuana**

### CODE REVIEW

File: guests/templates/checklist/band_merch_deadline.html [hunk 1]
```diff
@@ -230,6 +230,11 @@ <h4 class="modal-title" id="inventory_item_title_{{ item.id or type }}">{{ label
           setVisible('#coverage', isDedicated);
           setVisible('#rockisland', isRockIsland);
 
+          $('#rockisland input, #rockisland textarea, #rockisland select').prop('disabled', !isRockIsland);
```
Comment (RobRuana) [inline guests/templates/checklist/band_merch_deadline.html]: The hidden required form fields were preventing a band from selecting any option other than "Rock Island".

File: guests/templates/checklist/merch_deadline.html [hunk 2]
```diff
@@ -31,7 +31,7 @@
   <form method="post" action="merch" class="form-horizontal" role="form" enctype="multipart/form-data">
     {{ csrf_token() }}
     <input type="hidden" name="guest_id" value="{{ guest.id }}" />
-    <input type="hidden" name="id" value="{{ guest_merch.db_id }}" />
+    <input type="hidden" name="id" value="{{ guest_merch.db_id or 'None' }}" />
```
Comment (RobRuana) [inline guests/templates/checklist/merch_deadline.html]: If `guest_merch` is `None` then `db_id` is never called. We need this value to actually be the string `"None"`.

**Comment by RobRuana** [comment]
Merging now, cause people keep hitting this in production.

**Comment by kitsuta** [comment]
+1, belatedly!

**Pull request merged**